### PR TITLE
Update `LightningQubit` mid-circuit measurement preprocessing

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Changes needed after https://github.com/PennyLaneAI/pennylane/pull/5973 gets merged for correctness.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by mudit2812** [comment]
[sc-63618]

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 pip~=22.0
-git+https://github.com/PennyLaneAI/pennylane.git@master
+git+https://github.com/PennyLaneAI/pennylane.git@dos-grad-tests
```
Comment (mudit2812) [inline requirements-dev.txt:2]: Change before merge

```suggestion
git+https://github.com/PennyLaneAI/pennylane.git@master
```

**Review by AmintorDusko — APPROVED** [review]
Makes sense, thanks!

**Review by vincentmr — APPROVED** [review]
LGTM, thanks @mudit2812 . Don't forget to revert `reqs-dev.txt`.

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,9 @@
 * Add a Catalyst-specific wrapping class for Lightning Kokkos.
   [(#770)](https://github.com/PennyLaneAI/pennylane-lightning/pull/770)
 
+* Update `LightingQubit.preprocess` to work with changes to preprocessing for mid-circuit measurements.
+  [(#812)](https://github.com/PennyLaneAI/pennylane-lightning/pull/812)
```
Comment (maliasadi) [inline .github/CHANGELOG.md:43]: @mudit2812 We add new entries to the top of the list.

**Review by maliasadi — APPROVED** [review]
LGTM! 

Just don't forget those two minor comments before merge :)

**Pull request merged**